src/plugins/intel_gpu/include/intel_gpu/graph/network.hpp

```diff
@@ -36,7 +36,7 @@ struct network_output {
         // TODO: in_order queue doesn't create proper output event in some cases which leads to syncronization issues with user app
         // So call finish for associated stream to enusre that the output data is ready.
         if (do_sync) {
-            if (_stream->get_queue_type() == QueueTypes::in_order) {
+            if (_stream->get_queue_type() == QueueTypes::in_order || !_event) {
```

Comment:
When could event not be created for out-of-order queue?
If it's really possible, then this can add additional overhead in some cases, because currently some outputs can be processed asynchronously in `SyncInferRequest::wait()` function, however with this change it has to wait for all kernels to finish

Reply:
I think it happend when output primitive has CPU impl type which doesn't produce events anymore for barrier-based synchronization

Reply:
`make_output_event` seems to handle this case as well?